File: GameRanks/src/main/java/GameRanks/GameRanks/clientStruct/page/MainPageStruct.java
package GameRanks.GameRanks.clientStruct.page;

import GameRanks.GameRanks.clientStruct.element.DeveloperStruct;
import GameRanks.GameRanks.clientStruct.element.GameStruct;
import GameRanks.GameRanks.clientStruct.element.PublisherStruct;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class MainPageStruct {
    Iterable<GameStruct> topThreeGames;
    Iterable<PublisherStruct> topThreePublishers;
    Iterable<DeveloperStruct> topThreeDevelopers;
}

File: GameRanksClient/src/app/models/serverStructs/page/MainPageStruct.ts
import {GameStruct} from "../element/GameStruct";
import {PublisherStruct} from "../element/PublisherStruct";
import {DeveloperStruct} from "../element/DeveloperStruct";

export class MainPageStruct{
	topThreeGames: GameStruct[];
	topThreePublishers: PublisherStruct[];
	topThreeDevelopers: DeveloperStruct[];

	constructor(topThreeGames: GameStruct[], topThreePublishers: PublisherStruct[], topThreeDevelopers: DeveloperStruct[]){
		this.topThreeGames = topThreeGames;
		this.topThreePublishers = topThreePublishers;
		this.topThreeDevelopers = topThreeDevelopers;
	}
}
